src/main/native/CCM.c

```diff
@@ -267,12 +265,12 @@ JNIEXPORT jint JNICALL Java_com_ibm_crypto_plus_provider_ock_NativeInterface_do_
         handleIV_CCM(ivLen, keyLen, blockSize, J0Offset, (char*) iv, (char*) (parm_block + keyOffset), (char*) parm_block);
         memcpy(parm_block + counterValueOffset, parm_block + J0Offset + blockSize - 4, 4); // Add Counter Value
 
-        CCMzS390(input, output, aad, &len, &alen, parm_block, &mode);
+        CCMzS390((input != NULL) ? (input + inputOffset) : NULL, (output != NULL) ? (output + outputOffset) : NULL, aad, &len, &alen, parm_block, &mode);
 
-        if(inputJ != NULL) (*env)->ReleasePrimitiveArrayCritical(env, inputJ, input, 0);
-        if(outputJ != NULL) (*env)->ReleasePrimitiveArrayCritical(env, outputJ, output, 0);
+        if(inputJ != NULL) (*env)->ReleasePrimitiveArrayCritical(env, inputJ, (input != NULL) ? (input + inputOffset) : NULL, 0);
+        if(outputJ != NULL) (*env)->ReleasePrimitiveArrayCritical(env, outputJ, (output != NULL) ? (output + outputOffset) : NULL, 0);
 
-        return checkTagMismatch_CCM((char*) input, len, (char*) parm_block, tagOffset, tagLen);
+        return checkTagMismatch_CCM((input != NULL) ? (char*) (input + inputOffset) : NULL, len, (char*) parm_block, tagOffset, tagLen);
```

Comment:
How do we use `input` if it has already been released?

Reply:
Good point. This was likely working before but this definitely leaves room for possible memory corruption.